rfc/rfc-2/2 - Deletion Info Conversion.md

```diff
@@ -0,0 +1,221 @@
+
+## Proposers
+
+- @ashvin
+
+## Status
+
+GH Feature Request: https://github.com/apache/incubator-xtable/issues/339
+
+## Abstract
+Deletion vectors are one of the most popular features in table formats, allowing logical deletion of records, by marking
+them as deleted, without requiring rewriting of the data files to remove them physically. This document provides a 
+high-level proposal of deletion vector conversion in XTable, focusing on the current implementation of deletion vectors 
+in Delta Lake and Iceberg.
+
+## Background
+Deletion vectors feature in table formats like **_Delta Lake_** and **_Apache Iceberg_** are designed to enhance data 
+management efficiency. These vectors allow for the marking of deleted rows without physically removing them from the 
+data files, thereby optimizing operations such as `DELETE`, `UPDATE`, and `MERGE`. By maintaining a separate files of 
+deletions, deletion vectors enable faster query performance and reduce the need for costly data file rewrites. This 
+approach aligns with the merge-on-read paradigm, where changes are merged during read operations rather than write 
+operations, ensuring minimal disruption to ongoing data ingestion processes.
+
+The main types of delete vectors are _**equality deletes**_ and _**position deletes**_. 
+
+#### Equality deletes 
+Equality deletes allow for row-level deletions based on specific column values rather than their positions within a 
+file. This means that records in data files matching certain criteria are considered deleted without altering the 
+original data files. This deletion vector type is useful when identifying and storing the exact position of the row(s) 
+is expensive, but the content that identifies the row is clear. This feature is particularly useful in streaming use 
+cases where updates are frequent. Currently, Iceberg supports equality deletes, however, there is a proposal to replace 
+it in future versions.
+
+#### Position deletes
+Position deletes, on the other hand, specify the ordinals of records within a data file that should be considered 
+deleted. This method relies on identifying the exact position within the data file, thereby providing a more precise and
+direct approach to managing row-level deletions. Position deletes can be further categorized into two representations: 
+simple table representations and compressed bitmap representations.
+
+**_Simple table representations_** involve a straightforward structure typically stored in Parquet format, containing 
+columns like `file_path` and `pos`. The `file_path` indicates the path of the data file, while `pos` specifies the 
+ordinal of the record to be deleted. These files can optionally include the row data for equality deletes, aiding in 
+identifying the exact change which is particularly useful in CDC applications.
+
+A **_compressed bitmap representations_**, leverage a compressed bitmap to represent deletions. This method allows for a
```

Comment:
Are the binary deletion vector in iceberg V3 compatible with delta deletion vector at the file format level?
https://github.com/apache/iceberg/blob/main/format/spec.md#version-3-extended-types-and-capabilities

Would a direct copy work for those cases?

Reply:
Yes and no. The roaring bitmap byte array itself is compatible (same magic number, crc, etc), but the containing file isn't. In V3, deletion vectors are expected to be part of the puffin files, so Iceberg metadata can't directly reference Delta Lake deletion vectors files.